crates/iceberg-catalog/src/service/storage/mod.rs

```diff
@@ -59,29 +67,54 @@ impl StorageProfile {
             (StorageProfile::S3(this_profile), StorageProfile::S3(other_profile)) => {
                 this_profile.can_be_updated_with(other_profile)
             }
+            (StorageProfile::Az(this_profile), StorageProfile::Az(other_profile)) => {
+                this_profile.can_be_updated_with(other_profile)
+            }
             #[cfg(test)]
             (StorageProfile::Test(_), _) => Ok(()),
             #[cfg(test)]
             (_, StorageProfile::Test(_)) => Ok(()),
+            (_, _) => Err(ErrorModel::bad_request(
+                "Storage profiles are not compatible",
+                "StorageProfilesNotCompatible".to_string(),
+                None,
+            )
+            .append_details(&[
+                format!("This: {:?}", self.storage_type()),
+                format!("Other: {:?}", other.storage_type()),
+            ])
+            .into()),
         }
     }
 
     /// Create a new file IO instance for the storage profile.
     ///
     /// # Errors
     /// Fails if the underlying storage profile's file IO creation fails.
-    pub fn file_io(&self, secret: Option<&StorageCredential>) -> Result<iceberg::io::FileIO> {
+    pub fn file_io(
+        &self,
+        secret: Option<&StorageCredential>,
+        validating: bool,
```

Comment:
I don't like to add this too much just to set the error code. Couldn't we just throw the more used error code by default and overwrite on the outer scope if needed? It doesn't feel worth adding an argument.

Reply:
Ideally we'd not deal with error codes on this layer but just return an error which the respective handler then converts into the api response, relates to the revisit error handling issue, this module seems isolated enough to give it a try here.